### ISSUE TITLE
Remove `n_rows` and `levels_back` parameters

### ISSUE THREAD

**Issue by HenrikMettler** [Happy-Algorithms-League/hal-cgp#361]
For historical reasons, our genome has parameters`n_rows`, `levels_back`. However they offer little benefit in optimization and in actual applications these are never used. To simplify things and improve the user experience we should remove these parameters.

**Comment by HenrikMettler** [comment]
One question that arises, is whether we should also stop using the name `n_columns` and simply name it `n_hidden` since these things would become equal

**Comment by jakobj** [comment]
yes, i think that would be a good idea. `n_columns` may be difficult to understand without the option of adding rows :D
